rules/no-object-as-default.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+const MESSAGE_ID = 'noObjectAsDefault';
+
+const objectParameterSelector = [
+	'AssignmentPattern',
+	'[left.type="Identifier"]',
+	'[right.type="ObjectExpression"]',
+	'[right.properties.length>0]'
+].join('');
```

Comment:
If I remember correctly, the `AssignmentPattern` not only can use as parameter, maybe assignment default value? Anyway, we can use a stricter selector.

Reply:
Confirmed `const {foo={bar:1}} = baz;`